Source/Tools/FEXLoader/FEXLoader.cpp

```diff
@@ -150,47 +156,40 @@ void InterpreterHandler(fextl::string* Filename, const fextl::string& RootFS, fe
   // Is the file large enough for shebang
   if (ReadSize <= 2) {
     close(FD);
-    return;
+    return true;
   }
 
   // Handle shebang files
   if (Data[0] == '#' && Data[1] == '!') {
-    fextl::string InterpreterLine {Data.begin() + 2, // strip off "#!" prefix
-                                   std::find(Data.begin(), Data.end(), '\n')};
-    fextl::vector<fextl::string> ShebangArguments {};
+    std::string_view InterpreterLine {Data.begin() + 2, // strip off "#!" prefix
+                                      std::find(Data.begin(), Data.end(), '\n')};
+    fextl::vector<std::string_view> ShebangArguments {};
 
     // Shebang line can have a single argument
-    fextl::istringstream InterpreterSS(InterpreterLine);
-    fextl::string Argument;
-    while (std::getline(InterpreterSS, Argument, ' ')) {
-      if (Argument.empty()) {
-        continue;
+    using namespace std::literals;
+    auto Begin = InterpreterLine.begin();
+    auto ArgEnd = Begin;
+    auto End = InterpreterLine.end();
+    const auto Delim = " "sv;
+    for (; ArgEnd != End && Begin != End; Begin = ArgEnd + 1) {
+      // Find the end
+      ArgEnd = std::find_first_of(Begin, End, Delim.begin(), Delim.end());
+      if (Begin != ArgEnd) {
+        const auto View = std::string_view(Begin, ArgEnd - Begin);
+        if (!View.empty()) {
+          ShebangArguments.emplace_back(View);
+        }
       }
-      ShebangArguments.push_back(std::move(Argument));
     }
 
     // Executable argument
-    fextl::string& ShebangProgram = ShebangArguments[0];
-
-    // If the filename is absolute then prepend the rootfs
-    // If it is relative then don't append the rootfs
-    if (ShebangProgram[0] == '/') {
-      ShebangProgram = RootFS + ShebangProgram;
-    }
-    *Filename = ShebangProgram;
+    *Filename = ShebangArguments[0];
```

Comment:
Looks like this was already an issue before, but: This should use `.at(0)` instead so that malformed shebang lines are handled in any other way than UB.